agent/api/container/container.go

```diff
@@ -758,3 +758,19 @@ func (c *Container) ShouldCreateWithSSMSecret() bool {
 	}
 	return false
 }
+
+// MergeEnvironmentVariables appends additional envVarName:envVarValue pairs to
+// the the container's enviornment values structure
+func (c *Container) MergeEnvironmentVariables(secrets map[string]string) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	// don't assume that the environment variable map has been initialized by others
+	if c.Environment == nil {
+		c.Environment = make(map[string]string)
+	}
+
+	for k, v := range secrets {
+		c.Environment[k] = v
```

Comment:
@aws/aws-ecs-agent, @yumex93: how concerned should we be about user provided envvar names clobbering existing envvars? i'm not convinced we should be doing additional validation here.

Reply:
should do validation if there is an existing env var with the same key present?

Reply:
I already talked with Feng when he started to implement MTS, this validation is done there, so in theory, agent does not need to worry about this.

Reply:
hm. how would we handle that case? hm maybe log.Warnf + not add to the map?